### PR TITLE
Add optional sockets arg to connect peer

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Signed-off-by: Nitesh Balusu <niteshbalusu@icloud.com>

Sockets is an array which can undefined so this isn't a breaking change.

### CODE REVIEW

**Comment by alexbosworth** [comment]
if you have the socket why not just do addPeer

**Comment by niteshbalusu11** [comment]
> if you have the socket why not just do addPeer

helps to not do conditional checks to decide between addPeer and connectPeer. We could just use this for everything and pass sockets if we want to.

File: peers/connect_peer.js [hunk 1]
```diff
@@ -33,22 +35,41 @@ module.exports = ({id, lnd}, cbk) => {
...
         }
 
-        return getNode({
-          lnd,
-          is_omitting_channels: true,
-          public_key: id,
-        },
-        cbk);
+        // Exit early when sockets are provided
+        if (!!sockets && !!isArray(sockets) && !!sockets.length) {
```
Comment (alexbosworth) [inline peers/connect_peer.js:63]: if you do pass sockets, it should be validated to be a non empty array so you don't have to check here

File: peers/connect_peer.js [hunk 2]
```diff
@@ -58,7 +79,7 @@ module.exports = ({id, lnd}, cbk) => {
           return cbk();
         }
 
-        return asyncDetectSeries(getSockets.sockets, ({socket}, cbk) => {
+        return asyncDetectSeries(getSockets.sockets, (socket, cbk) => {
```
Comment (alexbosworth) [inline peers/connect_peer.js:82]: isn't there a mismatch here, like the result of getNode sockets is sockets->socket but the argument passed directly is an array of socket

**Comment by niteshbalusu11** [comment]
should be fixed now.

**Pull request merged**